python/ray/tests/gcp/test_gcp_node_provider.py

```diff
@@ -108,6 +116,80 @@ def test_convert_resources_to_urls_accelerators(test_case):
     assert modified_config["guestAccelerators"][0]["acceleratorType"] == result_accel
 
 
+def test_cpu_resource_returns_standard_command_runner():
+    mock_resource = MagicMock()
+
+    def __init__(self, provider_config: dict, cluster_name: str):
+        self.lock = RLock()
+        self.cached_nodes: Dict[str, GCPNode] = {}
+        self.resources: Dict[GCPNodeType, GCPResource] = {}
+        self.resources[GCPNodeType.COMPUTE] = mock_resource
+
+    with patch.object(GCPNodeProvider, "__init__", __init__):
+        node_provider = GCPNodeProvider({}, "")
+
+    args = {
+        "log_prefix": "test",
+        "node_id": "test-instance-compute",
+        "auth_config": auth_config,
+        "cluster_name": "test",
+        "process_runner": MockProcessRunner(),
+        "use_internal_ip": True,
+    }
+    command_runner = node_provider.get_command_runner(**args)
+    assert isinstance(command_runner, SSHCommandRunner)
+
+
+def test_cpu_resource_returns_standard_command_runner_docker():
+    mock_resource = MagicMock()
+
+    def __init__(self, provider_config: dict, cluster_name: str):
+        self.lock = RLock()
+        self.cached_nodes: Dict[str, GCPNode] = {}
+        self.resources: Dict[GCPNodeType, GCPResource] = {}
+        self.resources[GCPNodeType.COMPUTE] = mock_resource
+
+    with patch.object(GCPNodeProvider, "__init__", __init__):
+        node_provider = GCPNodeProvider({}, "")
+
+    args = {
+        "docker_config": {"container_name": "container"},
+        "log_prefix": "test",
+        "node_id": "test-instance-compute",
+        "auth_config": auth_config,
+        "cluster_name": "test",
+        "process_runner": MockProcessRunner(),
+        "use_internal_ip": True,
+    }
+    command_runner = node_provider.get_command_runner(**args)
+    assert isinstance(command_runner, DockerCommandRunner)
+
+
+def test_tpu_resource_returns_tpu_command_runner():
+    mock_resource = MagicMock()
+
+    def __init__(self, provider_config: dict, cluster_name: str):
+        self.lock = RLock()
+        self.cached_nodes: Dict[str, GCPNode] = {}
+        self.resources: Dict[GCPNodeType, GCPResource] = {}
+        self.resources[GCPNodeType.COMPUTE] = mock_resource
+        self.resources[GCPNodeType.TPU] = mock_resource
+
+    with patch.object(GCPNodeProvider, "__init__", __init__):
+        node_provider = GCPNodeProvider({}, "")
+
+    args = {
+        "log_prefix": "test",
+        "node_id": "test-instance-tpu",
+        "auth_config": auth_config,
+        "cluster_name": "test",
+        "process_runner": MockProcessRunner(),
+        "use_internal_ip": True,
+    }
+    command_runner = node_provider.get_command_runner(**args)
+    assert isinstance(command_runner, TPUCommandRunner)
+
+
```

Comment:
You could consider using `pytest.mark.parametrize` to reduce the repeated test code, that would also make it easy to add a "tpu+docker" test